frontend/manage_user.tsx

```diff
@@ -0,0 +1,220 @@
+  const [level, setLevel] = React.useState("owner")
+  const [genebank, setGenebank] = React.useState(0)
+  const [herds, setHerds] = React.useState([] as Herd[])
+  const [herd, setHerd] = React.useState(0)
+  const classes = useStyles();
+
+  const loadUser = (id: number) => {
+    setNew(false)
+    get(`/api/manage/user/${id}`).then(
+      data => data && setUser(data),
+      error => console.error(error)
+    );
+  }
+
+  React.useEffect(() => {
+    if (typeof id == 'number') {
+      loadUser(id)
+    } else if (id == 'new') {
+      setNew(true)
+      setUser({id: -1, email: '', validated: false, privileges: []})
+    }
+  }, [id])
+
+  React.useEffect(() => {
+    if (genebanks.length > 0) {
+      selectGenebank(genebanks[0].id)
+    }
+  }, [genebanks])
+
+  const selectGenebank = (gId: number) => {
+    setGenebank(gId);
+    let genebankData = genebanks.filter(g => g.id == gId);
+    if (genebankData.length > 0) {
+      setHerds(genebankData[0].herds)
+      if (herds.length > 0 && herds.filter(h => herd == h.id).length == 0) {
+        setHerd(herds[0].id)
+      }
+    } else {
+      setHerds([])
+      setHerd(0)
+    }
+  }
+
+  const submitForm = () => {
+    let postData = Object.assign({}, user);
+    delete postData["privileges"];
+    let protocol = isNew ? post : update;
+    protocol(`/api/manage/user/${id == 'new' ? 0 : id}`, postData).then(
+      data => {
+        switch (data.status) {
+          case "updated": console.info("updated."); break; // updated user
+          case "success":
+            loadData(["users"]).then(
+              success => loadUser(data.id)
+            )
+            break; // added user
+          default: console.warn("status:", data)// "failed" or other erro
+        }
+      },
+      error => console.error(error)
+    )
+  }
+
+  const updateRole = (operation: any) => {
+    post('/api/manage/role', operation).then(
+      data => {
+        switch (data.status) {
+          case "updated": loadUser(id)
+                          loadData(["users"])
+                          console.debug("updated successfully");
+                          break; // updated user
+          default: console.debug("error:", data)// "failed" or other error
+        }
+      },
+      error => console.error(error)
+    )
+  }
+
+  return <>
+    {user && <>
+      <h2>{isNew ? 'Ny användare' : user.email}</h2>
+      <form className={classes.form} noValidate autoComplete="off">
+          <TextField label='E-mail'
+                     type='email'
+                     className={classes.simpleField}
+                     value={user.email ?? undefined}
+                     onChange={e => setUser({...user, email: e.target.value})}
+            />
+          {isNew ? <TextField label='Lösenord'
+                      type='password'
+                      hidden={true}
+                      className={classes.simpleField}
+                      value={user.password ?? undefined}
```

Comment:
This needs to be `''` instead of `undefined` otherwise React gets confused if this is a controlled or uncontrolled component.

Reply:
ok! good to know! :)